pkg/flare/archive.go

```diff
@@ -141,18 +141,11 @@ func provideConfigDump(fb flaretypes.FlareBuilder) error {
 }
 
 func provideSystemProbe(fb flaretypes.FlareBuilder) error {
```

Comment:
@brycekahle offering a alternative suggestion. 

My understanding is the `pkg/flare/archive` is meant to deprecated eventually, as it is the old way of adding information to the flare. The new for is using flare provider via components or through Fx. 

One alternative to adding the system-probe flare files would be to include that information as part of the `cmd/agent/subcommands/run/command.go` Fx options

```go
fx.Options(
		flare.Module(flare.NewParams(
			defaultpaths.GetDistPath(),
			defaultpaths.PyChecksPath,
			defaultpaths.LogFile,
			defaultpaths.JmxLogFile,
			defaultpaths.DogstatsDLogFile,
			defaultpaths.StreamlogsLogFile,
		)),
		fx.Provide(func(sysprobeConfig sysprobeconfig.Component) flaretypes.Provider {
			return flaretypes.NewProvider(func(fb flaretypes.FlareBuilder) error {
				if sysprobeConfig.GetBool("system_probe_config.enabled") {
					_ = fb.AddFileFromFunc(filepath.Join("expvar", "system-probe"), getSystemProbeStats)
					_ = fb.AddFileFromFunc(filepath.Join("system-probe", "system_probe_telemetry.log"), getSystemProbeTelemetry)
				}

				if runtime.GOOS == "linux" {
					systemProbeConfigBPFDir := sysprobeConfig.GetString("system_probe_config.bpf_dir")
					if systemProbeConfigBPFDir != "" {
						fb.RegisterDirPerm(systemProbeConfigBPFDir)
					}

					sysprobeSocketLocation := getSystemProbeSocketPath()
					if sysprobeSocketLocation != "" {
						fb.RegisterDirPerm(filepath.Dir(sysprobeSocketLocation))
					}

					if sysprobeConfig.GetBool("system_probe_config.enabled") {
						_ = fb.AddFileFromFunc(filepath.Join("system-probe", "conntrack_cached.log"), getSystemProbeConntrackCached)
						_ = fb.AddFileFromFunc(filepath.Join("system-probe", "conntrack_host.log"), getSystemProbeConntrackHost)
						_ = fb.AddFileFromFunc(filepath.Join("system-probe", "ebpf_btf_loader.log"), getSystemProbeBTFLoaderInfo)
					}
				}
				return nil
			})
		}),
....
```

What would say to give that approach a try? 

Also, including @hush-hush in the conversation as he worked in the flare component 😄

Reply:
That would be better to reduce pkg/flare even more. WDYT about creating a dedicated components for "remote process" @GustavoCaso ? We could store all the logic for other process into `comp/core/flareproviders` or something similar ?

Since removing `pkg/flare/sysprobe` is already a huge win, thanks @brycekahle, this could be done in another PR

Reply:
>  WDYT about creating a dedicated components for "remote process" @GustavoCaso ? We could store all the logic for other process into comp/core/flareproviders or something similar ?

Absolutely. We have the `remoteagentregistry` component, which would allow to move all this logic for system-probe, trace-agent, security-agent, etc. 

Now that I think about it @brycekahle, lets do as @hush-hush mention and make this changes on a separate PR, I want to validate if system-probe could be a good candidate for using the remoteagentregistry component

Reply:
I'd also like to make system-probe generate a flare file, as the other processes do. Rather than requesting all the information over IPC. I was planning on tackling that in a future PR, so I can address your concerns then.